src/main.rs

```diff
@@ -397,6 +397,33 @@ fn main() -> anyhow::Result<()> {
         log::info!("Telemetry reporting disabled");
     }
 
+    //
+    // JWT RBAC
+    //
+    // Using HMAC-SHA256, recommended secret size is 32 bytes
+    const JWT_RECOMMENDED_SECRET_LENGTH: usize = 256 / 8;
+
+    // Log if JWT RBAC is enabled but no API key is set
+    if settings.service.jwt_rbac.unwrap_or_default() {
+        if settings
+            .service
+            .api_key
+            .clone()
+            .unwrap_or_default()
+            .is_empty()
+        {
+            log::error!("JWT RBAC enabled but no API key set")
+        // Log if JWT RAC is enabled, API key is set but smaller than recommended size for JWT secret
+        } else if settings.service.api_key.clone().unwrap_or_default().len()
+            < JWT_RECOMMENDED_SECRET_LENGTH
+        {
+            log::warn!(
+                "It is highly recommended to use an API key of {} bytes when JWT RBAC is enabled",
+                JWT_RECOMMENDED_SECRET_LENGTH
+            )
+        }
+    }
```

Comment:
It looks like this check only depends on `settings`. Might be better to move it to `validate_and_warn`

Reply:
I moved the code, let me know if it needs to be wrapped in a function or in some other way.